Custom post type for local groups

franei opened the pull request:
This contains a new custom post type for creating individual web pages for local groups.

#### New files:
- single-local_group.php: Template for a single local group
- group_5f1ad30f2103e.json: Set of custom fields for the WP admin
- local.scss: Styling for the template
- bird.svg: Background image

#### Edited files:
- functions.php: Define the new post type (and make some minor changes to the community group post type)
- package.json: Add a package ([owl-carousel](https://owlcarousel2.github.io/OwlCarousel2/)) for showing a picture gallery
- app.scss: Include styling for the template

Based on the discourse discussion and various requests for the tech team, the idea is that the template should be simple enough so that a new group can easily use it, but powerful enough to support a range of different sections that already established groups might want to use.

- Three fields are required: **Place**, **Region**, and **e-mail address**. Everything else is optional.
- Optional sections are **About us, Demands, Actions, Broadcast channels, Events, Volunteer positions,** and **Pictures**.
- Also optional are **social media links**, and **logo**.
- The template has a secondary navigation bar. If there is only the one required section (Contact), then the nav bar is hidden.
- The labels for the navigation (Contact, Over ons, Eisen, etc) cannot be changed by individual groups. The main headings within each section *can* be set individually.
- If there are no events for a group, the Events section will not be shown. The same is true for volunteer positions.

### Screenshots
#### Email only
![westerpark](https://user-images.githubusercontent.com/25393215/89738911-9ba05f80-da7c-11ea-9e24-5ad38d7a9615.png)
#### Events
![maastricht_events](https://user-images.githubusercontent.com/25393215/89738905-93482480-da7c-11ea-9e9d-e2382f0a5f25.png)
#### Demands
![groningen_eisen](https://user-images.githubusercontent.com/25393215/89738908-980cd880-da7c-11ea-92d5-fd8e2ab4ad5e.png)
#### About
![groningen](https://user-images.githubusercontent.com/25393215/89738912-9d6a2300-da7c-11ea-8e57-52b17132bee2.png)
#### Contact
![groningen_contact](https://user-images.githubusercontent.com/25393215/89738914-9fcc7d00-da7c-11ea-8d87-549c16e24822.png)
#### Photos
![lw](https://user-images.githubusercontent.com/25393215/89741321-90efc580-da90-11ea-83fc-4da32915b32c.png)

SamLubbers reviewed (changes requested):
## Feedback

WOOOOW @franei what a massive improvement to the website ❗ 😍   This is exactly the kind of feature that can really help grow and decentralize our movement ✊ ✊ ✊ 

It's so great that you have taken into account all previous discussions so that this page meets exactly the needs of local groups. I think people in local groups will really appreciate that they can choose their own sections and easily customize the content within those sections. On top of that, I think the design is fantastic, especially because of the ability to set a custom XR logo and navigate between tabs without having to refresh the page. 💯

Below and as inline comments I have left some minor change requests and some questions.

## Requested changes

* In the custom fields, the default URL for the extinction symbol doesn't seem to exist. The only one I found was on this URL. To fix this, you can set this as the default value `https://extinctionrebellion.nl/app/uploads/2019/10/X-symbol-richBlack.svg`, or eliminate the default value in the custom fields, as there is also a default file loaded from `/assets/images/XR-symbol.svg`

## Questions

* For what is the region data used? For grouping local groups in the future `/local` page? If that's the case, we can use [these 5 regions](https://wiki.extinctionrebellion.nl/Main/OrgRegionalSupportTeams) that XR NL uses to support local groups.
* Is it possible to have an WYSIWYG editor for the `action list > action > description` and `about > content`? I am asking because I think that right now it's only possible to add images with HTML, something that not many people know how to do.
* I see that some section headings are in Dutch and some in English. Is it possible to set the default value depending on the selected language? If not, we could even remove the default value for all except for `contact`, since the menu already indicates what is currently selected.
* What about making events and volunteer positions enabled by default? So we remove them from the custom fields, and instead render them only if they exist. 

## Other remarks

* The `birds.svg` image does not load on my local instance. I assume this occurs because the docker container does not use https, because when I load an image from a https website the image loads correctly. In the future it would be good to add https to the Docker image by default.
* I think it would be good to use the existing that events and volunteer positions have on each of their respective pages. That way it's easier for people to see that it refers to the same content. This is not urgent so I'll add it as new task. 

Thank you for the exceptionally commented pull request and for the immense amount of work required to make these great changes 💚

SamLubbers commented:
@franei by the way, I was very impressed to see the events and volunteer positions on the local group page. It's such a useful feature I hadn't even thought about 👌

franei commented on web/app/themes/xrnl/single-local_group.php:
```diff
@@ -0,0 +1,381 @@
+              </div>
+
+              <?php wp_reset_query(); ?>
+              <a class="btn btn-lg btn-black" href="<?php echo get_permalink(apply_filters('wpml_object_id', 548, 'post', true)) ?>">
```
The list now shows all events, so we can just drop that button

franei commented on web/app/themes/xrnl/single-local_group.php:
```diff
@@ -0,0 +1,381 @@
+  }
+?>
+
+<script src="/app/themes/xrnl/node_modules/bootstrap/dist/js/bootstrap.min.js"></script>
```
As far as I can tell bootstrap.js doesn't seem to be loaded from anywhere else

franei commented on web/app/themes/xrnl/single-local_group.php:
```diff
@@ -0,0 +1,381 @@
+  function getLocalEvents()
+  {
+    $events_query = array(
+      'posts_per_page' => 5,
+      'paged' => 1,
```
Done, showing all events now

franei commented on web/app/themes/xrnl/assets/sass/local.scss:
```diff
@@ -0,0 +1,277 @@
+        text-align: left;
+        border-bottom: 1px solid $xr-black;
+        .positions-info {
+          display: flex;
```
- I think it's ok with the latest browser versions, but anyway, there's not much harm in supporting older browsers (well except for some ugly code :))

SamLubbers replied:
I'm happy to see the internet is evolving haha. Instead of making our css very ugly, I added to the backlog a task for installing autoprefixer in the website build.

franei commented on web/app/themes/xrnl/assets/sass/local.scss:
```diff
@@ -0,0 +1,277 @@
+$xr-white:       map-get($theme-colors, 'white');
+$xr-primary:     map-get($theme-colors, 'primary');
+$xr-purple:      map-get($theme-colors, 'purple');
+$xr-lemon:       #f7ee6a;
+$xr-bright-pink: #cf6297;
```
Completely agree. (I think it would be nice to not have to call `map-get` every time for using a theme color. What do you think of defining the colours as individual variables instead of/in addition to an array?)

franei commented on web/app/themes/xrnl/assets/sass/local.scss:
```diff
@@ -0,0 +1,277 @@
+$logo-diameter: 200px;
+$logo-bg-diameter: 250px;
+$logo-bottom-shift: -50px;
+$logo-bg-bottom-shift: $logo-bottom-shift - ( ($logo-bg-diameter - $logo-diameter)/2 );
```
😁

franei commented:
#### Requested changes
- Yes, good point. I've removed the redundant default value from the custom field.

#### Questions
- Regions: Yes, this was meant for grouping on /local page. Maybe not necessary right now, but once we have hundreds of local groups..
- WYSIWYG editors. 2 options:
	1. Make them WYSIWYG editors, and the content goes into a `<div>` instead of a `<p>`.
	2. Add an image field to the custom field set, and put that into an inline `<img>`.
Somewhat leaning towards option 2. Actually I'm wondering if it would be wise to strip html tags from all user input (maybe except things like bold or italic text), so that we don't end up with a soup of html tags and bootstrap classes in the WP admin, like on some other pages 🙃. But maybe we can discuss this further.
- Headings: I don't think it's possible to adjust the default value depending on the language of the post that's being added. But you're right, headings may be unnecessary, given the navbar. Let's just leave them empty by default.
- Events and volunteer roles: I agree, let's make them enabled by default. If we remove the whole CF though, it wouldn't be possible to add custom text above the lists. Do you think it's unnecessary? I have no strong opinion either way.

#### Other remarks
- bird.svg not loading: It might be because `file_get_contents()` is checking for https. I've added a parameter so that hopefully it won't check when `WP_ENV` is set to "development".
- Use the existing ?? for events and volunteer positions: I'm guessing the missing word is styling 😄 Happy to change it either way!

franei commented on web/app/themes/xrnl/single-local_group.php:
```diff
@@ -0,0 +1,381 @@
+    ]);
+    $local_roles = array();
+    $n_local_roles = 0;
+    $group_name = strtolower('XR ' . get_field('place'));
```
What do you think of adding a "Group name" field, like this?
<img width="866" alt="image" src="https://user-images.githubusercontent.com/25393215/91087863-6f2a3d00-e651-11ea-8a65-46634d96e19f.png">

The advantage over using the post title would be that we can add instructions and make it required.

SamLubbers replied:
Yes that's perfect, especially because of the clear instructions of what is expected in the group name and place field 💯 

Wow, I also just saw that the name for the group name already has `XR` as a predefined prefix 🤤

franei commented on web/app/themes/xrnl/single-local_group.php:
```diff
@@ -0,0 +1,381 @@
+<div class="local-group">
+
+  <div class="lg-hero-container px-3 py-5 pb-5 text-center">
+    <h1 class="display-2">XR <?php the_field('place'); ?></h1>
```
It's now using the new `group_name` field (see above).

SamLubbers reviewed (commented):
@franei thanks for all the quick changes 🙌  (and sorry for not being able to get back to you earlier 😓 )

* The picture custom field is a good replacement for the WYSIWYG editor. I think once local groups start using it we'll get more feedback on the style of the about us and actions page. 
* Let's keep the description under the events and roles custom fields, so that local groups can share any information they need (e.g. all events are cancelled due to corona / if you don't find an interesting role sending us an email at...)
* birds.svg still refuses to load in http haha. Since the current docker image also has issues with other pictures let's leave this for now and wait until we have a docker image with TLS enabled by default.
